core/modules/replica/AbortTransactionJob.cc

```diff
@@ -0,0 +1,344 @@
+
+    for (auto&& request: _requests) {
+        request->cancel();
+    }
+}
+
+
+void AbortTransactionJob::notify(util::Lock const& lock) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << __func__);
+
+    notifyDefaultImpl<AbortTransactionJob>(lock, _onFinish);
+}
+
+
+void AbortTransactionJob::_onRequestFinish(SqlDeleteTablePartitionRequest::Ptr const& request) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << __func__
+         << "  worker=" << request->worker()
+         << " _requests.size()=" << _requests.size()
+         << " _numLaunched=" << _numLaunched
+         << " _numFinished=" << _numFinished
+         << " _numSuccess=" << _numSuccess);
+
+    if (state() == State::FINISHED) return;
+
+    util::Lock lock(_mtx, context() + string(__func__));
+
+    if (state() == State::FINISHED) return;
+
+    ++_numFinished;
+    if (request->extendedState() == Request::SUCCESS) ++_numSuccess;
+
+    // Replace the finished job with a new one unless the input set is empty
+    if (0 == _submitNextBatch(lock, 1)) {
+
+        if (_numFinished == _numLaunched) {
+
+            LOGS(_log, LOG_LVL_DEBUG, context() << __func__
+                 << "  worker=" << request->worker()
+                 << " _requests.size()=" << _requests.size());
+
+            // Harvest results from all jobs (regardless of their completion status)
+            // and be done.
+
+            for (auto&& ptr: _requests) {
+                LOGS(_log, LOG_LVL_DEBUG, context() << __func__
+                     << " requestId=" << ptr->id()
+                     << " targetRequestParams=" << ptr->targetRequestParams());
+                auto const table = ptr->targetRequestParams().table;
+                _resultData.completed [ptr->worker()][table] = ptr->extendedState() == Request::SUCCESS;
+                _resultData.resultSets[ptr->worker()][table] = ptr->responseData();
+            }
+            finish(lock, _numSuccess == _numLaunched ? ExtendedState::SUCCESS
+                                                     : ExtendedState::FAILED);
+        }
+    }
+}
+
+
+size_t AbortTransactionJob::_submitNextBatch(util::Lock const& lock,
+                                             size_t const maxRequests) {
+
+    if (0 == maxRequests) return maxRequests;
+
+    auto self = shared_from_base<AbortTransactionJob>();
+
+    size_t batchSize = 0;
+    for (size_t i = 0; i < maxRequests; ++i) {
+
+        // Scan the input queue and find a worker which has the longest list
+        // of tables to be processed. Then submit a request for any table
+        // of that worker.
+        
+        size_t maxNumTables = 0; 
+        string candidateWorker;
+        for (auto&& entry: _worker2tables) {
+            string const& worker = entry.first;
+            size_t const numTables = entry.second.size();
+            if (numTables > maxNumTables) {
+                maxNumTables = numTables;
+                candidateWorker = worker;
+            }
+        }
```

Comment:
May be not. The current algorithm iterates over elements of:
```
std::map<std::string,std::list<std::string>> _worker2tables
```
And to do what I need to do here I would have to write either a non-trivial functor or a lambda function. All together this would increase the number of lines on top of what people usually do when using `std::max_element` on vectors of numbers. The current implementation is straightforward. 
But thank you for the suggestion! I will keep this algorithm in mind.

Reply:
I believe you can do `std::max_element` in two lines of code with a relatively trivial lambda. This implementation is straightforward but you have to read more lines of code to understand what it does.